contracts/PangolinExchange.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.2;
+
+import "@pangolindex/exchange-contracts/contracts/pangolin-periphery/interfaces/IPangolinRouter.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./IAssetsExchange.sol";
+import "./SupportedAssets.sol";
+
+/**
+ * @title PangolinExchange
+ * @dev Contract allows user to invest into an ERC20 token
+ * This implementation uses the Pangolin DEX
+ */
+contract PangolinExchange is Ownable, IAssetsExchange {
+  /* ========= STATE VARIABLES ========= */
+  IPangolinRouter pangolinRouter;
```

Comment:
maybe `dexAddress` or sth like that?

Reply:
I think the current name better describes what this Object actually is. Especially that in the Uniswap/Pangolin the Router is not the only contract that you may want to interact with.